File: backend/services/comment-service/internal/model/comment.go
package model

import "time"

// Comment represents a comment on a post.
type Comment struct {
	ID             string    `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	PostID         string    `gorm:"type:uuid;not null;index:idx_comments_post" json:"post_id"`
	AuthorID       string    `gorm:"type:uuid;not null;index" json:"author_id"`
	AuthorNickname string    `gorm:"type:varchar(100);not null" json:"author_nickname"`
	AuthorAvatar   string    `gorm:"type:text;default:''" json:"author_avatar"`
	Content        string    `gorm:"type:text;not null" json:"content"`
	IsAIGenerated  bool      `gorm:"default:false" json:"is_ai_generated"`
	CreatedAt      time.Time `gorm:"autoCreateTime" json:"created_at"`
}

func (Comment) TableName() string {
	return "comments"
}

File: frontend/lib/domain/models/comment.dart
/// Comment model matching the backend Comment entity.
class Comment {
  final String id;
  final String postId;
  final String authorId;
  final String authorNickname;
  final String authorAvatar;
  final String content;
  final bool isAiGenerated;
  final DateTime createdAt;

  const Comment({
    required this.id,
    required this.postId,
    required this.authorId,
    required this.authorNickname,
    this.authorAvatar = '',
    required this.content,
    this.isAiGenerated = false,
    required this.createdAt,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'] as String,
      postId: json['post_id'] as String,
      authorId: json['author_id'] as String,
      authorNickname: json['author_nickname'] as String,
      authorAvatar: json['author_avatar'] as String? ?? '',
      content: json['content'] as String,
      isAiGenerated: json['is_ai_generated'] as bool? ?? false,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}
